csharp/ql/src/semmle/code/csharp/commons/Util.qll

```diff
@@ -5,7 +5,11 @@ import csharp
 /** A `Main` method. */
 class MainMethod extends Method {
   MainMethod() {
-    this.hasName("Main") and
+    (
+      this.hasName("Main")
+      or
+      this.hasQualifiedName("<Program>$.<Main>$")
```

Comment:
`this.hasQualifiedName("<Program>$", "<Main>$")` is better because it avoids computing `hasQualifiedName/1`, which will construct a lot of strings.